tests/filter.json

```diff
@@ -407,6 +407,23 @@
         {"a":"c", "d": "f"}
       ]
     },
+    {
+      "name": "exists and exists, data false",
+      "selector" : "$[?@.a&&@.b]",
+      "document" : [{"a": false, "b": false}],
```

Comment:
Can we add an item here that doesn't match?

Reply:
This test case has been updated to include non-matching data.